### PR TITLE
doc: example for splitWhen and splitAfter, #25468

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

File: akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/Split.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    // #splitWhen
+    Source.range(1, 100)
+        .throttle(1, Duration.ofMillis(100))
+        .map(elem -> new Pair<>(elem, Instant.now()))
+        .statefulMapConcat(
+            new Creator<Function<Pair<Integer, Instant>, Iterable<Pair<Integer, Boolean>>>>() {
+              // stateful decision in statefulMapConcat
+              // keep track of time bucket (one per second)
+              LocalDateTime currentTimeBucket =
+                  LocalDateTime.ofInstant(Instant.ofEpochMilli(0), ZoneOffset.UTC);
```
Comment (patriknw) [inline akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/Split.java]: wrong placement

File: akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/Split.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  public static void splitWhenExample(String[] args) {
+    ActorSystem system = ActorSystem.create();
+
+    // #splitWhen
+    Source.range(1, 100)
+        .throttle(1, Duration.ofMillis(100))
+        .map(elem -> new Pair<>(elem, Instant.now()))
+        .statefulMapConcat(
+            () -> {
+              return new Function<Pair<Integer, Instant>, Iterable<Pair<Integer, Boolean>>>() {
```
Comment (patriknw) [inline akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/Split.java]: `new` needed because lambda can't change variables

Comment (johanandren) [reply akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/Split.java]: I've seen that a common Java-land idiom for that is a one element array container. But I think this is just as good (or better).

**Review by johanandren — APPROVED** [review]
Nice, LGTM!

**Review by chbatey — APPROVED** [review]
Nice example

**Pull request merged**